modules/Sources/Features/Home/Home.swift

```diff
@@ -163,10 +180,22 @@ public struct Home: Reducer {
             case let .cantStartSync(error):
                 state.destination = .alert(.cantStartSync(error))
                 return .none
+            case .fetchLatestFiatPrice:
+                guard !state.isFiatLoaded, state.preferredCurrency != . off else { return .none }
+                state.isFiatLoaded = true
+                return .run { [preferredCurrency = state.preferredCurrency] send in
+                    try? await send(.latestFiatResponse(fiatPriceClient.getZcashPrice(preferredCurrency)))
+                }
+            case let .latestFiatResponse(price):
+                userStoredPreferences.setLatestFiatPrice(price)
```

Comment:
[Blocking] 
is this a "caching" workaround?

How would you determine that the value is not fresh anymore?

Reply:
It wasn't really so much as a "caching" scheme as it was a "shared global" scheme so I don't have to pass it around everywhere. But in chat with Adi, we really want this to be tied to the app session, so this should truly just be stored in non-persistent memory. Will change to resolve this.